enhancements/machine-api/remediation-history.md

```diff
@@ -0,0 +1,168 @@
+condition triggered the remediation, and the timestamps of
+- when the triggering condition was detected
+- when remediation was started
+- when the node was succesfully fenced (deleted / powered off)
+- when the node is healthy again.
+
+This information can be displayed in the UI in a table on a machine healthcheck details page.
+
+Example MHC:
+
+```yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineHealthCheck
+metadata:
+  annotations:
+    machine.openshift.io/remediation-strategy: external-baremetal
+  creationTimestamp: "2020-11-24T16:31:22Z"
+spec:
+  maxUnhealthy: 100%
+  nodeStartupTimeout: 60m
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-machine-role: worker
+  unhealthyConditions:
+  - status: "False"
+    timeout: 20s
+    type: Ready
+  - status: Unknown
+    timeout: 20s
+    type: Ready
+status:
+  conditions:
+  - lastTransitionTime: "2020-11-24T17:17:32Z"
+    status: "True"
+    type: RemediationAllowed
+  currentHealthy: 2
+  expectedMachines: 2
+  remediationHistory:
+  - conditionStatus: Unknown
+    conditionType: Ready
+    detected: "2020-11-24T17:33:43Z"
+    fenced: "2020-11-24T17:34:35Z"
+    finished: "2020-11-24T17:35:09Z"
+    remediationType: external
+    started: "2020-11-24T17:34:04Z"
+    targetKind: Node
+    targetName: worker-1
+  remediationsAllowed: 2
+```
+
+Example UI: TBD (copy from "KNI Node & Host Management Designs" doc?)
+
+### User Stories
+
+#### Story 1
+
+As a cluster administrator I want to have an overview over when and why nodes failed and were remediated, in order
+to detect patterns, and to be able to troubleshoot possible root causes.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The machine healthcheck controller [0] is responsible for triggering remediations based on machine health checks and node
+conditions. This will also be the place to track remediations:
+
+- in the `needsRemediation` func:
+  - for failed machines and unhealthy node conditions: track detection
+  - for missing nodes after remdiation start: track successful fencing
+  - for healthy nodes: track finished remdiation
+    
+- in the `remediate` and `remediationStrategyExternal` funcs: track remediation start
+
+[0] https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/machinehealthcheck/machinehealthcheck_controller.go
+
+### Risks and Mitigations
+
+We are not aware of any risks.
+
+## Design Details
+
+### Open Questions [optional]
+
+TBD
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+TBD
+
+### Version Skew Strategy
+
+TBD
+
+## Implementation History
+
+TBD
+
+## Drawbacks
+
+## Alternatives
+
+- Using events
+  
+  The UI already displays events (in its own tab), but they aren't stored for very long, which will make it harder to see which nodes
+  are having recurring issues.
```

Comment:
What is not very long, can we quantify this at all?

Reply:
done